src/mpid/ch4/netmod/ofi/ofi_rma.h

```diff
@@ -519,6 +519,11 @@ static inline int MPIDI_OFI_do_put(const void *origin_addr,
         msg.rma_iov_count = tout;
         MPIDI_OFI_CALL_RETRY2(MPIDI_OFI_INIT_CHUNK_CONTEXT(win, sigreq),
                               fi_writemsg(ep, &msg, flags), rdma_write);
+
+        /* By default, progress is called only during fence, which significantly
+         * slows down the RMA operations for large non-contiguous data. Adding manual
+         * progress helps improve the performance. */
+        MPIDI_OFI_win_counter_check(win);
     }
```

Comment:
As the counter checking is actually for manual progress, can we rename the function to something like `MPIDI_OFI_rma_progress` ? For example, it can be also implemented by internally calling `fi_cq_read`.

Reply:
I changed the function to be called `MPIDI_OFI_win_trigger_rma_progress`.